tests/subsys/power/power_mgmt/src/dummy_driver.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/printk.h>
+#include <zephyr/types.h>
+#include "dummy_driver.h"
+
+static uint32_t device_power_state;
+
+static int dummy_open(struct device *dev)
+{
+	return device_pm_get_sync(dev);
+}
+
+static int dummy_close(struct device *dev)
+{
+	return device_pm_put_sync(dev);
+}
+
+static uint32_t dummy_get_power_state(struct device *dev)
+{
+	return device_power_state;
+}
+
+static int dummy_suspend(struct device *dev)
+{
+	device_power_state = DEVICE_PM_SUSPEND_STATE;
+	return 0;
+}
+
+static int dummy_resume_from_suspend(struct device *dev)
+{
+	device_power_state = DEVICE_PM_ACTIVE_STATE;
+	return 0;
+}
+
+static int dummy_device_pm_ctrl(struct device *dev, uint32_t ctrl_command,
+				void *context, device_pm_cb cb, void *arg)
+{
+	int ret = 0;
+
+	switch (ctrl_command) {
+	case DEVICE_PM_SET_POWER_STATE:
+		if (*((uint32_t *)context) == DEVICE_PM_ACTIVE_STATE) {
+			ret = dummy_resume_from_suspend(dev);
+		} else {
+			ret = dummy_suspend(dev);
+		}
+		break;
+	case DEVICE_PM_GET_POWER_STATE:
+		*((uint32_t *)context) = dummy_get_power_state(dev);
+		break;
+	default:
+		ret = -EINVAL;
+
+	}
+
+	if (cb)
```

Comment:
This `if` statement needs to be contained in brackets, and needs to handle the else condition as well to avoid unexpected/undefined behavior.

Reply:
Thank you @jenmwms , updated.